apps/examples/security_test/security_test_main.c

```diff
@@ -0,0 +1,1582 @@
+			break;
+		}
+
+		/* negative case 1 : exceed max size */
+		memset(output, 0, SEE_MAX_DATA_SIZE);
+		ret = see_write_secure_storage(input, inlen[2], j);
+		ret += see_read_secure_storage(output, &inlen[2], j);
+		if (!ret) {
+			test_result |= 0x1;
+			see_selfprintf(" fail\n");
+			break;
+		} else {
+			see_selfprintf(" success\n");
+		}
+
+		inlen[0] = 18;
+		inlen[1] = SEE_MAX_DATA_SIZE;
+		inlen[2] = SEE_MAX_DATA_SIZE + 1;
+	}
+
+	see_selfprintf("[%d] ", j);
+
+	ret = 0;
+	memset(output, 0, 200);
+	ret = see_write_secure_storage(input, inlen[0], SSTORAGE_DATA_SLOT_INDEX_MAX);
+	ret += see_read_secure_storage(output, &inlen[0], SSTORAGE_DATA_SLOT_INDEX_MAX);
+	if (!ret) {
+		test_result |= 0x1;
+		see_selfprintf(" fail\n");
+	} else {
+		see_selfprintf(" success\n");
+	}
+
+	return test_result;
+}
+
+#define SEE_GENERATE_KEY_TC 16
+
+unsigned int TC_see_generate_key(void)
+{
+	int ret = 0;
+	int i = 0;
+	int j = 0;
+	unsigned int test_result = 0;
+
+	unsigned int genkey_input[SEE_GENERATE_KEY_TC][3] = { {AES_KEY, 16, SEE_OK}, {AES_KEY, 24, SEE_OK}, {AES_KEY, 32, SEE_OK},
+		{AES_KEY, 48, SEE_OK}, {AES_KEY, 64, SEE_OK}, {AES_KEY, 80, SEE_ERROR},
+		{HMAC_KEY, 64, SEE_OK}, {HMAC_KEY, 128, SEE_OK}, {HMAC_KEY, 129, SEE_ERROR},
+		{ECC_KEY_BP256, 0, SEE_OK}, {ECC_KEY_NIST192, 0, SEE_OK},
+		{ECC_KEY_NIST224, 0, SEE_OK}, {ECC_KEY_NIST256, 0, SEE_OK},
+		{ECC_KEY_NIST384, 0, SEE_OK}, {ECC_KEY_NIST512, 0, SEE_OK},
+		{0xffff, 0, SEE_INVALID_INPUT_PARAMS}
+	};
+
+	printf("----------------------------------------\n");
+	printf("TESTCASE : see_generate_key\n");
+	printf("----------------------------------------\n");
+
+	for (j = 0; j < SEE_GENERATE_KEY_TC; j++) {
+		for (i = 0; i < 9; i++) {
+			see_selfprintf("[%d] ", j * 9 + i);
+			ret = see_generate_key(genkey_input[j][0], i, genkey_input[j][1], 0);
+			if (i == 8 && ret) {
+				see_selfprintf(" success\n");
+			} else if (ret != genkey_input[j][2]) {
+				test_result |= 0x4;
+				see_selfprintf(" fail\n");
+				continue;
+			} else {
+				see_selfprintf(" success\n");
+			}
+		}
+	}
+
+	isp_clear(0);
+
+	unsigned int genkey_input_rsa[4][3] = { {RSA_KEY_1024, 17, 0}, {RSA_KEY_2048, 17, 0},
+		{RSA_KEY_1024, 0, 0}, {RSA_KEY_2048, 0, 0}
+	};
+
+	for (j = 0; j < 4; j++) {
+		see_selfprintf("[%d] ", j);
+
+		ret = see_generate_key(genkey_input_rsa[j][0], 0, 0, genkey_input_rsa[j][1]);
+		if (ret != genkey_input_rsa[j][2]) {
+			test_result |= 0x4;
+			see_selfprintf(" fail\n");
+		} else {
+			see_selfprintf(" success\n");
+		}
+	}
+	isp_clear(0);
+	return test_result;
+
+}
+
+unsigned int TC_see_setup_remove_key(void)
+{
+	int ret = 0;
+	int i = 0;
+	int j = 0;
+	unsigned int test_result = 0;
+	unsigned int key_size[5] = { 32, 128, 1212, 1212, 256 };
+	unsigned int key_type[5] = { SECURE_STORAGE_TYPE_KEY_AES,
+								 SECURE_STORAGE_TYPE_KEY_HMAC, SECURE_STORAGE_TYPE_KEY_RSA,
+								 SECURE_STORAGE_TYPE_KEY_DH, SECURE_STORAGE_TYPE_KEY_ECC
+							   };
+	unsigned char key[1280];
```

Comment:
Huge buff size, you should check stack size of this app very carefully.

Reply:
Good point, I'll change static declarations of some large buffers into malloc.